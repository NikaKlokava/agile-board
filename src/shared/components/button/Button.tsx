import cl from "./button.module.css";
import classes from "classnames";

export const Button = ({ text, className, ...props }: any) => {
  return (
    <button {...props} className={classes(cl.button, className)}>
      {text}
    </button>
  );
};
