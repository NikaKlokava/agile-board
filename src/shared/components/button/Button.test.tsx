import renderer from "react-test-renderer";
import { Button } from "./Button";

describe("Test the Button component", () => {
  test("The Button renders correctly", () => {
    const buttonSnap = renderer
      .create(<Button text={"Test Btn"} withIcon={false} testid={"test"} />)
      .toJSON();
    expect(buttonSnap).toMatchSnapshot();
  });
});
