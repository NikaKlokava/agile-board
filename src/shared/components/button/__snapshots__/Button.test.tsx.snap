// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Button component The Button renders correctly 1`] = `
<div
  className="button"
>
  <div
    className="button_text"
  >
    Test Btn
  </div>
</div>
`;

exports[`Test the Input component The Input renders correctly 1`] = `
<div
  className="button"
>
  <div
    className="button_text"
  >
    Test Btn
  </div>
</div>
`;
